### PR TITLE
SFR-391 Implement design / component.

### PR DESCRIPTION
**Pull request by nachocs**
After test review from Chris, this commit solve the differences found:
Added Search Footer.
Fixed some margins
First, next, previous links shouldn't be underlined
Page shown 1 of total
Format numbers
Added line down view all editions
Fix font in usa-table
Added functionality in Select to accept value and label.
SearchFooter is very similar to SearchHeader but I think it's better to keep them separated for better maintenance

### CODE REVIEW

File: src/app/components/SearchResults/ResultsMetadata.jsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ const ResultsMetadata = ({ searchQuery, metadata: { total } }) => {
     lastElement = total;
   }
   if (total > 0) {
-    message = `Viewing ${firstElement} - ${lastElement} of ${total} items`;
+    message = `Viewing ${firstElement.toLocaleString()} - ${lastElement.toLocaleString()} of ${total.toLocaleString()} items`;
```
Comment (mwbenowitz) [inline src/app/components/SearchResults/ResultsMetadata.jsx]: I didn't know you could localize numbers in this way!

File: src/client/styles/components/EditionsList.scss [hunk 2]
```diff
@@ -1,6 +1,6 @@
 .nypl-editions-table {
   width: 100%;
-  margin: 0 0 20px 0;
+  margin: 0;
```
Comment (mwbenowitz) [inline src/client/styles/components/EditionsList.scss]: The table is now running into the NYPL footer in some (all?) instances with the `mainContent` padding removed as well. Not sure where that padding should live, but it seems like there should be something there

File: src/client/styles/main.scss [hunk 3]
```diff
@@ -56,7 +58,7 @@ body {
 }
 
 .main-content {
-  margin: 0 auto 60px;
+  margin: 0 auto;
```
Comment (mwbenowitz) [inline src/client/styles/main.scss]: Related to the comment above, this is causing an issue on the work detail page

File: src/app/components/SearchResults/ResultsMetadata.jsx (hunk 1 above)
Comment (gkallenberg) [reply src/app/components/SearchResults/ResultsMetadata.jsx]: Definitely a cool feature. Specific to the browser's locale? If so, do all browsers support this?

File: src/app/components/SearchResults/SearchResults.jsx [hunk 4]
```diff
@@ -3,6 +3,7 @@ import PropTypes from 'prop-types';
 import ResultsMetadata from './ResultsMetadata';
 import ResultsList from './ResultsList';
 import SearchHeader from './SearchHeader';
+import SearchFooter from './SearchFooter';
```
Comment (gkallenberg) [inline src/app/components/SearchResults/SearchResults.jsx]: Wondering if there couldn't be some shared components between SearchHeader and SearchFooter treatments. Are there specific things that are different now for both or would this purely be for future implementation of unique elements for this content bar? If you could share some things, that would help with minimizing repetition of code.

File: src/client/styles/components/EditionsList.scss (hunk 2 above)
Comment (nachocs) [reply src/client/styles/components/EditionsList.scss]: That's true, I didn't see it. I just made a commit to solve that

**Comment by nachocs** [comment]
> This looks good with just a few minor comments/questions. I left a few in the code about some `padding` issues I noticed on the work detail page.
> 
> Also a few other general thoughts:
> 
> * Did Jack ask for the new divider line between works in the search results to be heavier than the others? Can't remember what he said in standup yesterday
> * This divider line is missing when there isn't a "View all" link, is that unnecessary? unsure of what's in the designs
> * I like the footer paging, but the "Page" label sitting on its own line looks off to me. Do we need it? Obviously a minor question/one for the design team.

In the designs right now, the divider is 1 px and only shows when there is "view all". 
Yes that "Page" looks off, it's in the design like that.

File: src/app/components/SearchResults/SearchResults.jsx (hunk 4 above)
Comment (nachocs) [reply src/app/components/SearchResults/SearchResults.jsx]: I began using the SearchHeader with some flag, but quickly found out that it was a very artificial  difference that could lead to confusion in the future. There are some common components that could be separated

File: src/app/components/SearchResults/ResultsMetadata.jsx (hunk 1 above)
Comment (nachocs) [reply src/app/components/SearchResults/ResultsMetadata.jsx]: Yes, it's specific to the browser locale, although we can get it fixed to 'en-US' if we want. 
Supported in all browsers

Comment (gkallenberg) [reply src/app/components/SearchResults/ResultsMetadata.jsx]: No need to fix it to our way. It's always nice to see things in one's own particular idiom.

**Pull request merged**